src/main/java/org/fogbowcloud/saps/engine/core/archiver/storage/swift/SwiftPermanentStorage.java

```diff
@@ -22,43 +31,54 @@
 
 public class SwiftPermanentStorage implements PermanentStorage {
 
-
     private static final Logger LOGGER = Logger.getLogger(SwiftPermanentStorage.class);
     private static final String SWIFT_TASK_STAGE_DIR_PATTERN = "%s" + File.separator + "%s" + File.separator + "%s";
     private static final int MAX_ARCHIVE_TRIES = 1;
     private static final int MAX_SWIFT_UPLOAD_TRIES = 2;
+    private static final String TEMP_DIR_URL = "%s?temp_url_sig=%s&temp_url_expires=%s";
+    private static final String HMAC_SHA1_ALGORITHM = "HmacSHA1";
 
     private final SwiftAPIClient swiftAPIClient;
-    //FIXME Remove properties field and add new variables
-    private final Properties properties;
     private final String nfsTempStoragePath;
     private final String containerName;
+    private final String swiftKey;
+    private final String tasksDirName;
+    private final String debugTasksDirName;
     private final boolean debugMode;
 
+
     public SwiftPermanentStorage(Properties properties, SwiftAPIClient swiftAPIClient) throws PermanentStorageException {
         if (!checkProperties(properties))
             throw new PermanentStorageException("Error on validate the file. Missing properties for start Swift Permanent Storage.");
```

Comment:
Add TODO to change the type of exception thrown to `InvalidPropertyException` or change it in this PR

Reply:
Done.